gitman/git.py

```diff
@@ -259,6 +259,13 @@ def get_branch():
     return git("rev-parse", "--abbrev-ref", "HEAD", _show=False)[0]
 
 
+def get_object_rev(object_name):
+    """Get the revision associated with the object specified."""
+    log_results = git("log", "-n", "1", object_name)
```

Comment:
Would it make sense for this to capture exceptions and return a tuple of `commit_sha, valid` to be used by the caller? I think we want to avoid the CLI displaying exception stacktraces, which can happen now:

```
$ poetry run gitman lock gitman_1@main

Locking dependencies...

$ cd /Users/Browning/Code/gitman/demo

  $ cd gitman_1
  $ git log -n 1 main
  fatal: ambiguous argument 'main': unknown revision or path not in the working tree.
  Use '--' to separate paths from revisions, like this:
  'git <command> [<revision>...] -- [<file>...]'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/Browning/Code/gitman/gitman/cli.py", line 239, in main
    _run_command(function, args, kwargs)
  File "/Users/Browning/Code/gitman/gitman/cli.py", line 325, in _run_command
    success = function(*args, **kwargs)
  File "/Users/Browning/Code/gitman/gitman/decorators.py", line 9, in wrapped
    result = function(*args, **kwargs)
  File "/Users/Browning/Code/gitman/gitman/commands.py", line 264, in lock
    count = config.lock_dependencies(*names, obey_existing=False)
  File "/Users/Browning/Code/gitman/gitman/models/config.py", line 230, in lock_dependencies
    source_locked = source.lock(skip_changes=skip_changes, rev=rev)
  File "/Users/Browning/Code/gitman/gitman/models/source.py", line 236, in lock
    rev = git.get_object_rev(rev)
  File "/Users/Browning/Code/gitman/gitman/git.py", line 264, in get_object_rev
    log_results = git("log", "-n", "1", object_name)
  File "/Users/Browning/Code/gitman/gitman/git.py", line 16, in git
    return call("git", *args, **kwargs)
  File "/Users/Browning/Code/gitman/gitman/shell.py", line 94, in call
    raise ShellError(message, program=program, output=complete_output)
gitman.exceptions.ShellError: An external program call failed.

In working directory: /Users/Browning/Code/gitman/demo/gitman_1

The following command produced a non-zero return code:

$ git log -n 1 mainfatal: ambiguous argument 'main': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

Reply:
Agreed, this should catch ShellErrors, log the error and then return `None` rather than a tuple.

Reply:
Passing `_ignore=True` should handle this for you: https://github.com/jacebrowning/gitman/blob/c00f7c698640d6f4eef848c592321b484a81b1cd/gitman/shell.py#L15-L25